### PR TITLE
[392705] Fancy Xbase/Xtend syntax coloring

### PR DESCRIPTION
**Pull request by sailingKieler**
See https://bugs.eclipse.org/bugs/show_bug.cgi?id=392705

The test class `XtendHighlightingCalculatorExtendedColoringTest.xtend` is just a stub for the moment, I plan to add more tests shortly.

After a bit of massage the existing tests are running, so I'm confident that I didn't break anything.

### CODE REVIEW

File: plugins/org.eclipse.xtext.xbase.ide/src/org/eclipse/xtext/xbase/ide/highlighting/XbaseHighlightingCalculator.java [hunk 1]
```diff
@@ -161,8 +168,47 @@ protected void highlightReferenceJvmType(IHighlightedPositionAcceptor acceptor,
...
+		if (resolvedReferencedObject instanceof JvmTypeParameter) {
+			// may happen in cast expressions
+			highlightFeature(acceptor, referencer, reference, TYPE_VARIABLE);
+			
+		} else if (referencer instanceof JvmParameterizedTypeReference
+					&& (referencersContainingFeature == TypesPackage.Literals.JVM_PARAMETERIZED_TYPE_REFERENCE__ARGUMENTS
+							|| referencersContainingFeature == TypesPackage.Literals.JVM_TYPE_CONSTRAINT__TYPE_REFERENCE
+							|| referencersContainingFeature == XbasePackage.Literals.XABSTRACT_FEATURE_CALL__TYPE_ARGUMENTS)) {
+			// case 1: 'referencer' is a type reference within the arguments reference of another (parameterized) type reference
+			//  'referencer' definitely a type argument and to be colored as such
```
Comment (sailingKieler) [inline plugins/org.eclipse.xtext.xbase.ide/src/org/eclipse/xtext/xbase/ide/highlighting/XbaseHighlightingCalculator.java]: 'referencer' IS...

File: plugins/org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/highlighting/XbaseHighlightingConfiguration.java [hunk 2]
```diff
@@ -77,6 +105,72 @@ public TextStyle deprecatedMembers(){
...
+	
+	public TextStyle classes() {
+		return defaultTextStyle().copy();
+	}
+	
+	public TextStyle abstractClasses() {
+		return classes().copy();
+	}
+	
+	public TextStyle staticFinalField(){
```
Comment (akosyakov) [inline plugins/org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/highlighting/XbaseHighlightingConfiguration.java]: a missing whitespace between () and {
the same true for some methods below

**Comment by dustContributor** [comment]
Its happening! :smile:

**Comment by akosyakov** [comment]
Please merge, so we can test it with Neon on Monday.

**Pull request merged**